tool/generator/lib/src/projection/utils.dart

```diff
@@ -341,6 +345,217 @@ String parseGenericTypeIdentifierName(TypeIdentifier typeIdentifier) {
+    return 'enum($enumName;$enumSignature)';
+  }
+
+  if (typeProjection.isWinRTStruct) {
+    final structName = ti.name;
+    final fields = ti.type!.fields
+        .map((field) => parseTypeIdentifierSignature(field.typeIdentifier));
+    return 'struct($structName;${fields.join(';')})';
+  }
+
+  if (typeProjection.isInterface) return ti.type!.guid!.toLowerCase();
+
+  if (typeProjection.isClass) {
+    final className = ti.name;
+    final defaultInterface = ti.type!.interfaces.first;
+    if (defaultInterface.typeSpec != null) {
+      final defaultInterfaceSignature =
+          parseTypeIdentifierSignature(defaultInterface.typeSpec!);
+      return 'rc($className;$defaultInterfaceSignature)';
+    }
+
+    final defaultInterfaceSignature = parseTypeDefSignature(defaultInterface);
+    return 'rc($className;$defaultInterfaceSignature)';
+  }
+
+  return primitiveTypeSignatureFromBaseType(ti.baseType);
+}
+
+/// Returns the type signature for the given generic [typeIdentifier].
+String parseGenericTypeIdentifierSignature(TypeIdentifier typeIdentifier) {
+  if (typeIdentifier.baseType != BaseType.genericTypeModifier) {
+    throw ArgumentError('Expected a generic type identifier.');
+  }
+
+  final parentTypeGuid = typeIdentifier.type!.guid!.toLowerCase();
+
+  if (typeIdentifier.type?.genericParams.length == 2) {
+    final firstArgSignature =
+        parseTypeIdentifierSignature(typeIdentifier.typeArg!);
+    final secondArgSignature =
+        parseTypeIdentifierSignature(typeIdentifier.typeArg!.typeArg!);
+    return 'pinterface($parentTypeGuid;$firstArgSignature;$secondArgSignature)';
+  }
+
+  return 'pinterface($parentTypeGuid;${parseTypeIdentifierSignature(typeIdentifier.typeArg!)})';
+}
+
+const wrtPinterfaceNamespace = [
+  0x11,
+  0xf4,
+  0x7a,
+  0xd5,
+  0x7b,
+  0x73,
+  0x42,
+  0xc0,
+  0xab,
+  0xae,
+  0x87,
+  0x8b,
+  0x1e,
+  0x16,
+  0xad,
+  0xee,
+];
+
+/// Returns the IID for the given [signature].
+///
+/// To learn more about this, please see https://learn.microsoft.com/en-us/uwp/winrt-cref/winrt-type-system#guid-generation-for-parameterized-types
+String iidFromSignature(String signature) {
+  final signatureInBytes = const Utf8Encoder().convert(signature);
+  final data = [...wrtPinterfaceNamespace, ...signatureInBytes];
+  final sha1Bytes = sha1.convert(data).bytes;
+
+  final first = Int8List(4)
+    ..[0] = sha1Bytes[0]
+    ..[1] = sha1Bytes[1]
+    ..[2] = sha1Bytes[2]
+    ..[3] = sha1Bytes[3];
+  final firstNumber = first.buffer.asByteData().getUint32(0);
+
+  final second = Int8List(2)
+    ..[0] = sha1Bytes[4]
+    ..[1] = sha1Bytes[5];
+  final secondNumber = second.buffer.asByteData().getUint16(0);
+
+  final third = Int8List(2)
+    ..[0] = sha1Bytes[6]
+    ..[1] = sha1Bytes[7];
+  var thirdNumber = third.buffer.asByteData().getUint16(0);
+  thirdNumber = (thirdNumber & 0x0fff) | (5 << 12);
+
+  final fourthNumber = (sha1Bytes[8] & 0x3f) | 0x80;
```

Comment:
These parts are heavily inspired by the Rust/WinRT projection https://github.com/microsoft/windows-rs/blob/1df6f66312ea7cb1e5e08eb7f9f0522b364cc267/crates/libs/windows/src/core/guid.rs#L44C53-L59